### ISSUE TITLE
bootstrap script doesn't enforce version arg in the second run

### ISSUE THREAD

**Issue by thedebugger** [saltstack/salt-bootstrap#1117]
### Steps to Reproduce Issue
```
[vagrant@dock001 ~]$ curl -o bootstrap-salt.sh -L https://bootstrap.saltstack.com/develop
[vagrant@dock001 ~]$ chmod +x bootstrap-salt.sh
[vagrant@dock001 ~]$ salt-call --version
salt-call 2015.5.10 (Lithium)
[vagrant@dock001 ~]$ ./bootstrap-salt.sh stable 2017.7.0
...
```

Try installing the old version.
```
[vagrant@dock001 ~]$ sudo ./bootstrap-salt.sh stable 2015.5.10
 *  INFO: Running version: 2017.05.24
 *  INFO: Executed by: /bin/sh
 *  INFO: Command line: './bootstrap-salt.sh stable 2015.5.10'
 *  WARN: Running the unstable version of bootstrap-salt.sh

 *  INFO: System Information:
 *  INFO:   CPU:          GenuineIntel
 *  INFO:   CPU Arch:     x86_64
 *  INFO:   OS Name:      Linux
 *  INFO:   OS Version:   3.10.0-327.36.1.el7.x86_64
 *  INFO:   Distribution: CentOS 7.2

 *  INFO: Installing minion
 *  INFO: Found function install_centos_stable_deps
 *  INFO: Found function config_salt
 *  INFO: Found function preseed_master
 *  INFO: Found function install_centos_stable
 *  INFO: Found function install_centos_stable_post
 *  INFO: Found function install_centos_restart_daemons
 *  INFO: Found function daemons_running
 *  INFO: Found function install_centos_check_services
 *  INFO: Running install_centos_stable_deps()
Package yum-utils-1.1.31-34.el7.noarch already installed and latest version
Package chkconfig-1.3.61-5.el7_2.1.x86_64 already installed and latest version
Package PyYAML-3.11-1.el7.x86_64 already installed and latest version
Nothing to do
 *  INFO: Running install_centos_stable()
Package salt-minion-2017.7.0-1.el7.noarch already installed and latest version
Nothing to do
 *  INFO: Running install_centos_stable_post()
 *  INFO: Running install_centos_check_services()
 *  INFO: Running install_centos_restart_daemons()
 *  INFO: Running daemons_running()
*  INFO: Salt installed!
[vagrant@dock001 ~]$ echo $?
0
[vagrant@dock001 ~]$ salt-call --version
salt-call 2017.7.0 (Nitrogen)
```

### Versions and Systems
```
[vagrant@dock001 ~]$ ./bootstrap-salt.sh -v
./bootstrap-salt.sh -- Version 2017.05.24
```
```
[vagrant@dock001 ~]$ uname -a
Linux dock001.dev.creditkarma.com 3.10.0-327.36.1.el7.x86_64 #1 SMP Sun Sep 18 13:04:29 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

### Expected result
It should install the specified version or error out if it can't install

Let me know if you need more information.

**Comment by rallytime** [comment]
Hi @thedebugger - I can reproduce this as well. Thanks for letting me know.

In the mean time as a work-around, you can use the `git` option to install from any release tag: `./bootstrap-salt.sh git v2015.5.10`.

**Comment by thedebugger** [comment]
Hi @rallytime, i have tried doing that i think the git install doesn't install the dependencies (new issue?). I'm seeing the following error when bootstrap script tries to start minion
```
sudo ./bootstrap-salt.sh -D git 'v2017.7.0'

Traceback (most recent call last):
  File "/usr/bin/salt-minion", line 26, in <module>
    salt_minion()
  File "/usr/lib/python2.6/site-packages/salt/scripts.py", line 161, in salt_minion
    import salt.cli.daemons
  File "/usr/lib/python2.6/site-packages/salt/cli/daemons.py", line 47, in <module>
    from salt.utils import parsers, ip_bracket
  File "/usr/lib/python2.6/site-packages/salt/utils/parsers.py", line 28, in <module>
    import salt.config as config
  File "/usr/lib/python2.6/site-packages/salt/config/__init__.py", line 100, in <module>
    _DFLT_IPC_WBUFFER = _gather_buffer_space() * .5
  File "/usr/lib/python2.6/site-packages/salt/config/__init__.py", line 90, in _gather_buffer_space
    import salt.grains.core
  File "/usr/lib/python2.6/site-packages/salt/grains/core.py", line 47, in <module>
    import salt.utils.dns
  File "/usr/lib/python2.6/site-packages/salt/utils/dns.py", line 28, in <module>
    import salt.modules.cmdmod
  File "/usr/lib/python2.6/site-packages/salt/modules/cmdmod.py", line 34, in <module>
    import salt.utils.templates
  File "/usr/lib/python2.6/site-packages/salt/utils/templates.py", line 31, in <module>
    import salt.utils.http
  File "/usr/lib/python2.6/site-packages/salt/utils/http.py", line 41, in <module>
    import salt.loader
  File "/usr/lib/python2.6/site-packages/salt/loader.py", line 26, in <module>
    import salt.utils.event
  File "/usr/lib/python2.6/site-packages/salt/utils/event.py", line 82, in <module>
    import salt.transport.ipc
  File "/usr/lib/python2.6/site-packages/salt/transport/ipc.py", line 21, in <module>
    from tornado.locks import Semaphore
```
Here are the full logs -- https://gist.github.com/thedebugger/f8dd7bb3dfe5c2ff1685a2d1f4410a66

I also tried v2015.5.10 and it fails because it can't import `m2crypto`

Let me know if you need more information

**Comment by rallytime** [comment]
Ok, @thedebugger there are a couple of different issues here. I'll try to address them each:

- For 2017.7.0, we no longer support python2.6. It's 2.7 and newer only, so that's likely why you're seeing that stacktrace you posted above. If you need 2017.7.0 on CentOS 6 (which defaults to Python 2.6), I would suggest using the `-x` and `-y` options that were added recently to handle this. These install Python 2.7 on CentOS 6: `salt-bootstrap.sh -y -x python2.7 git v2017.7.0`.
- The second issue is that `v2015.5.10` is an old version of Salt and as Salt's dependencies change, so do the packages that get installed with salt-bootstrap. `m2crypto` is no longer a dependency of Salt starting with version 2015.8.0. As such, the `m2crypto` installation was removed from salt-bootstrap in PR #951.

If you still would like to install a version from 2015.5, I would suggest using the bootstrap script from a release before the m2crypto package was removed: https://github.com/saltstack/salt-bootstrap/tree/v2016.08.16.

Another option is to use the current stable bootstrap script to install a newer version of Salt in the `2016.11.x` range.

**Comment by thedebugger** [comment]
@rallytime that makes sense and thanks for explaining in detail. I'm using the vagrant and vagrant salt plugin doesn't allow to change bootstrap to a specific version. I guess it is okay - I can use shell provisioner and pull the specific version. One thing we can do, in bootstrap script, is error out when one tries to install a version that is no longer supported by the script.

**Comment by rallytime** [comment]
Ok @thedebugger, after looking into this some more, I have discovered what the issue is and this is expected behavior. 

When you run the bootstrap script the first time with `stable 2017.7`, the repo file in `/etc/yum.repos.d/` gets set up to point to `latest`:
```
# cat /etc/yum.repos.d/saltstack.repo
[saltstack]
name=SaltStack latest Release Channel for RHEL/CentOS $releasever
baseurl=https://repo.saltstack.com/yum/redhat/7/$basearch/latest/
skip_if_unavailable=True
gpgcheck=1
gpgkey=https://repo.saltstack.com/yum/redhat/7/$basearch/latest/SALTSTACK-GPG-KEY.pub
enabled=1
enabled_metadata=1
```
Then, when you run the script again with `stable 2016.11`, the bootstrap script already has a repo file in place, so it installs from that repo file. 

Basically, to fix this issue, you need to remove the old repo file and remove the salt packages that were installed previously, and clear out your yum cache. Then you can run the script again.
